src/librustc_errors/emitter.rs

```diff
@@ -347,8 +347,7 @@ impl EmitterWriter {
 
         // Sort the annotations by (start, end col)
         let mut annotations = line.annotations.clone();
-        annotations.sort();
-        annotations.reverse();
+        annotations.sort_by(|a,b| b.start_col.cmp(&a.start_col));
```

Comment:
I agree that a comment would be good. I personally have no idea why this yields the correct result. =) But the change looks good!